src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+    this.kind = 'herbivore';
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.kind = 'carnivore';
+  }
+
+  bite(animal) {
+    if (animal.kind === 'herbivore' && !animal.hidden) {
+      animal.health -= 50;
+    }
+
+    if (!animal.health) {
+      Animal.alive.sort((a, b) => b.health - a.health).pop();
```

Comment:
very smart way to remove a dead animal, but i think you should change the !animal.health condition. because it will only be true if heals is 0, if an animal has 30 health and then gets bitten, health would be -20 and your condition will not work

Reply:
И тут ты тоже прав! Слишком заточено под одно условие.